pkg/dynamicconfig/table_test.go

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dynamicconfig
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/hivetest"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/health/types"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/testutils"
+)
+
+var (
+	cmName         = "cilium-config"
+	namespace      = "kube-system"
+	dummyConfigMap = map[string]string{"key1": "value1", "key2": "value2"}
+)
+
+func TestWatchKey(t *testing.T) {
+	_, db, dct, _ := fixture(t)
+
+	done := make(chan bool)
+	key := "key"
+	lowPrioritySource := "lowPriority"
+	value := "value"
+	newValue := "newValue"
+
+	upsertDummyEntry(db, dct, key, lowPrioritySource, value)
+
+	e, w := WatchKey(db.ReadTxn(), dct, key)
+
+	if e.Key.Name != key || e.Value != value || e.Key.Source != lowPrioritySource {
+		t.Errorf("Entry mismatch for key %v: expected (key=%v, source=%v, value=%v), but got (key=%v, source=%v, value=%v)", key, key, lowPrioritySource, value, e.Key, e.Key.Source, e.Value)
+	}
+
+	go waitFor(func() {
+		<-w
+		e, _ := WatchKey(db.ReadTxn(), dct, key)
+		if e.Key.Name != key || e.Value != newValue || e.Key.Source != cmName {
+			t.Errorf("Entry mismatch for key %v: expected (key=%v, source=%v, value=%v), but got (key=%v, source=%v, value=%v)", key, key, lowPrioritySource, value, e.Key, e.Key.Source, e.Value)
```

Comment:
Do we need this `waitFor` construction? This should be doable all from a single goroutine, e.g. you do `upsertDummyEntry` and then you block on the watch channel and validate. It's also not advisable to use `testing.T` from other goroutines.

Reply:
Yes you are right. removed the waitFor. Though I've kept the select in case the test is failing to timeout after 2 seconds? Do you see a better way to do it?